### PR TITLE
Change default folder in dacpac and schema compare extensions

### LINKED ISSUE #4457: Dacpac: File location should default to opened folder if available
1. Open a folder in Azure Data Studio
2. Launch Extract action in dac wizard
3. Look at file location

Expected:
- Shows dacpac save location under current folder, which is how most stuff in VSCode works
Actual:
- Saves under user root, which is really not ideal  as it's not where anyone would look.

image

### PR DESCRIPTION
**Pull request by kisantia**
This changes the default folder in the dacpac and schema compare extensions to be the root path opened in the file explorer if a folder is open, otherwise the default is the user's home folder. 

Fixes  #4457

### CODE REVIEW

File: extensions/dacpac/src/wizard/api/dacFxConfigPage.ts [hunk 1]
```diff
@@ -148,7 +149,8 @@ export abstract class DacFxConfigPage extends BasePage {
 	protected generateFilePath(): string {
 		let now = new Date();
 		let datetime = now.getFullYear() + '-' + (now.getMonth() + 1) + '-' + now.getDate() + '-' + now.getHours() + '-' + now.getMinutes();
-		return path.join(os.homedir(), this.model.database + '-' + datetime + this.fileExtension);
+		let rootPath = vscode.workspace.rootPath ? vscode.workspace.rootPath : os.homedir();
+		return path.join(rootPath, this.model.database + '-' + datetime + this.fileExtension);
```
Comment (pensivebrian) [inline extensions/dacpac/src/wizard/api/dacFxConfigPage.ts]: Can the database name contain illegal file characters?

File: extensions/dacpac/src/wizard/pages/deployActionPage.ts [hunk 2]
```diff
@@ -169,7 +169,8 @@ export class DeployActionPage extends DacFxConfigPage {
 	private setDefaultScriptFilePath(): void {
 		let now = new Date();
 		let datetime = now.getFullYear() + '-' + (now.getMonth() + 1) + '-' + now.getDate() + '-' + now.getHours() + '-' + now.getMinutes();
-		this.fileTextBox.value = path.join(os.homedir(), this.model.database + '_UpgradeDACScript_' + datetime + '.sql');
+		let rootPath = vscode.workspace.rootPath ? vscode.workspace.rootPath : os.homedir();
```
Comment (pensivebrian) [inline extensions/dacpac/src/wizard/pages/deployActionPage.ts]: This line of code is repeated a lot. Can you put it in a method to centralize the logic?

File: extensions/dacpac/src/wizard/api/dacFxConfigPage.ts (hunk 1 above)
Comment (kisantia) [reply extensions/dacpac/src/wizard/api/dacFxConfigPage.ts]: Opened an issue #5216. Will fix in a separate PR

File: extensions/dacpac/src/wizard/pages/deployActionPage.ts (hunk 2 above)
Comment (kisantia) [reply extensions/dacpac/src/wizard/pages/deployActionPage.ts]: I moved the one in the dacpac extension to a method in the base page class

File: extensions/dacpac/src/wizard/api/dacFxConfigPage.ts [hunk 3]
```diff
@@ -146,9 +147,17 @@ export abstract class DacFxConfigPage extends BasePage {
 	}
 
 	protected generateFilePath(): string {
```
Comment (pensivebrian) [inline extensions/dacpac/src/wizard/api/dacFxConfigPage.ts]: nit naming: generateFilePathFromDatabaseAndTimestamp()

**Pull request merged**